snuba/query/__init__.py

```diff
@@ -63,6 +65,10 @@ class Query(ABC, Generic[TDataSource]):
     Each of the node is an AST expression or a sequence of AST
     expressions.
 
+    This class can represent either a query on a simple data source
+    (entity or relational source) or a composite query with nested
+    queries, joins, or a combination of them.
+
```

Comment:
I suspect I will make a specific subclass to allow processing of composite queries when I will start doing join optimizations, I will break it then since I am not yet sure of which kind of methods it will be useful to expose.